game/configuration.cc

```diff
@@ -268,13 +273,20 @@ void writeConfig(Game& game, bool system) {
 		else if (type == "uint") entryNode->set_attribute("value",std::to_string(item.ui()));
 		else if (type == "bool") entryNode->set_attribute("value", item.b() ? "true" : "false");
 		else if (type == "float") entryNode->set_attribute("value", std::to_string(item.f()));
-		else if (item.getType() == "string") xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(item.s());
-		else if (item.getType() == "string_list") {
-			for (auto const& str : item.sl()) xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(str);
+		else if (type == "string") xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(item.s());
+		else if (type == "string_list") {
+			for (auto const& str : item.sl()) {
+				xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(str);
+			}
 		}
-		else if (item.getType() == "option_list") {
+		else if (type == "option_list") {
 			//TODO: Write selected also (as attribute?)
-			for (auto const& str : item.ol()) xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(str);
+			auto const selectedValue = item.so();
+			xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(selectedValue);
+
+			for (auto const& str : item.ol())
+				if(str != selectedValue)
+					xmlpp::add_child_element(entryNode, "stringvalue")->add_child_text(str);
```

Comment:
Missing some braces here might aswell early continue to reduce nesting.
Or even better if such a functionality is available, already pre-filter the collection before looping as we only want strings which isn't the current selected value. in C# we use linq (in fluent syntax) a lot which does exactly that

Reply:
Added the braces. The other point needs ranges, else we bloat the code.